src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -439,6 +444,30 @@ def encode_prompt(
 
         return prompt_embeds, negative_prompt_embeds
 
+    # Note (sayakpaul): Name it this way to not mess up with other functions like _encode_image()
+    # common in imag2image pipelines.
+    def encode_image_ip_adapter(self, image, device, num_images_per_prompt):
+        dtype = next(self.image_encoder.parameters()).dtype
+
+        if not isinstance(image, torch.Tensor):
+            image = self.feature_extractor(image, return_tensors="pt").pixel_values
+
+        image = image.to(device=device, dtype=dtype)
+        image_embeds = self.image_encoder(image).image_embeds
+        projected_image_embeds = self.image_projection(image_embeds)
```

Comment:
I didn't do this because I don't think we should attach weights to pipelines 

I think it is better use the `imageProjection` layer in UNET  cc @patrickvonplaten here

Reply:
the `encode_image` I added for IP-adapter is no difference from `encode_image` in existing pipelines - same way as different pipelines have may have slightly different implementation of `encode_prompts` but essentially they do same thing

Reply:
> I didn't do this because I don't think we should attach weights to pipelines

Fair point. 

> the encode_image I added for IP-adapter is no difference from encode_image in existing pipelines - same way as different pipelines have may have slightly different implementation of encode_prompts but essentially they do same thing

Yeah with your implementation that doesn't use the projection module, it makes sense to name the function as `encode_image()` as that is exactly what it is doing.

Reply:
> I didn't do this because I don't think we should attach weights to pipelines. I think it is better use the imageProjection layer in UNET cc @patrickvonplaten here

Agree